warehouse/integrations/azure-synapse/azure-synapse.go

```diff
@@ -459,15 +512,19 @@ func (as *AzureSynapse) ProcessColumnValue(
 	case model.BooleanDataType:
 		return strconv.ParseBool(value)
 	case model.StringDataType:
-		if len(value) > stringLengthLimit {
-			value = value[:stringLengthLimit]
+		if varcharLength == varcharMaxLength {
+			return value, nil
+		}
+		maxStringLength := max(varcharLength, varcharDefaultLength)
+		if len(value) > maxStringLength {
+			value = value[:maxStringLength]
 		}
 		if !hasDiacritics(value) {
 			return value, nil
 		} else {
 			byteArr := str2ucs2(value)
-			if len(byteArr) > stringLengthLimit {
-				byteArr = byteArr[:stringLengthLimit]
+			if len(byteArr) > maxStringLength {
```

Comment:
Given that we have already truncated `value` above to `maxStringLength`, is it still possible for `byteArr` length to be more than `maxStringLength` ?

Reply:
Since we are already slicing to maxStringLength, it's length will not be more than maxStringLength.

Reply:
Then can we get rid of `if len(byteArr) > maxStringLength` code since it will always be false ?

Reply:
`str2ucs2` function converts a **UTF-8** string to **UCS-2 encoding** using **UTF-16 encoding** where it's length will increase.